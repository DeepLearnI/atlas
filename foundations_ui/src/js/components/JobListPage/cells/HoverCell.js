import React, { Component } from 'react';
import PropTypes from 'prop-types';

class HoverCell extends Component {
  render() {
    const { textToRender, onMouseLeave } = this.props;
    return (
      <div onMouseLeave={onMouseLeave} className="job-cell-hover" data-class="hover-cell">
        {textToRender}
      </div>
    );
  }
}

HoverCell.propTypes = {
  textToRender: PropTypes.object,
  onMouseLeave: PropTypes.func,
};

HoverCell.defaultProps = {
  textToRender: <p />,
  onMouseLeave: () => {},
};

export default HoverCell;
