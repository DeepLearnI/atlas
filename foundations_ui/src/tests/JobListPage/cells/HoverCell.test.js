import React from 'react';
import ReactDOM from 'react-dom';
import HoverCell from '../../../js/components/JobListPage/cells/HoverCell';
import { shallow, mount } from 'enzyme';
import configureTests from '../../setupTests';

configureTests();

it('Shallow Renders HoverCell', () => {
    const wrapper = shallow(<HoverCell/>);
    expect(wrapper).toMatchSnapshot();
  });