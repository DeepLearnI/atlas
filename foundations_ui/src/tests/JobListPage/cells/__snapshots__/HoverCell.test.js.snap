// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders HoverCell 1`] = `
<div
  className="job-cell-hover"
  data-class="hover-cell"
  onMouseLeave={[Function]}
>
  <p />
</div>
`;
